ehrql/backends/tpp.py

```diff
@@ -360,3 +360,21 @@ class TPPBackend(BaseBackend):
             FROM ISARIC_New
         """
     )
+
+    open_prompt = QueryTable(
+        """
+        SELECT
+            Patient_ID AS patient_id,
+            CodedEvent_ID AS coded_event_id,
+            CASE
+                WHEN CodeSystemId = 0 THEN ConceptId
+            END AS snomedct_concept_id,
+            CASE
+                WHEN CodeSystemId = 2 THEN ConceptId
+            END AS snomedy_concept_id,
```

Comment:
(Not a comment on your SQL, this is just a natural place to hang the question!)

Do we know if each survey answer appears as two rows, once with a ConceptId from each system? Or do some questions always use one system and other questions always use another?

Reply:
A response doesn't have a PK. Nevertheless, we can infer that one response is represented by one row -- rather than, say, two rows; one for each coding system -- by grouping by `Consultation_ID` (a survey's PK) and counting the number of rows. If one response were represented by two rows, we'd expect the number of rows to be even in all cases. It isn't.

Reply:
See opensafely/airmid-short-data-report#11

Reply:
See opensafely/airmid-short-data-report#12